Source/WebCore/platform/graphics/Path.cpp

```diff
@@ -152,7 +152,7 @@ void Path::addLineTo(const FloatPoint& point)
     else if (auto moveTo = asSingleMoveTo())
         m_data = PathSegment(PathDataLine { moveTo->point, point });
     else
-        ensureImpl().addLineTo(point);
+        ensureImpl().add(PathLineTo { point });
```

Comment:
If the Impl is `PathCG`, this adds the overhead of copying the point to structure here. Then getting the address of the point in the structure in `PathCG::add(PathMoveTo)`.

So I do not see the benefit of this change so far. Maybe in #[10594](https://github.com/WebKit/WebKit/pull/10594), I will see the benefit of it.

Reply:
>If the Impl is PathCG, this adds the overhead of copying the point to structure here. 

No, I don't think there's an additional overhead.
Before, the pointer read to registers happened in PathCG::moveTo.
After,  the pointer read to registers happens in Path::addLineTo.
It's the same amount of reads.

> Then getting the address of the point in the structure in PathCG::add(PathMoveTo).

No, there's no pointers anymore in PathCG::add. It's pass by value.
I think the values are already in the registers.

Reply:
> So I do not see the benefit of this change so far. Maybe in #10594, I will see the benefit of it.

The intended benefit of having uniform name and passing the PathSegment::Data values as values:
It is done to be able to write polymorphic functions (==templates) over PathSegment. 

For PathImpl::add over moveTo,bezierCurveTo,arcTo, it's evident: we are able to deconstruct PathSegment and apply the deconstructed variants in uniform fashion, and we don't need to further deconstruct the various PathMoveTo, ..., structs into moveTo(a,b,c,) calls redundantly.